src/adaptors/mod.rs

```diff
@@ -1110,6 +1110,46 @@ impl<I, J> Iterator for Flatten<I, J>
 
 }
 
+/// An iterator adapter to apply `Into` conversion to each element.
+///
+/// See [`.map_into()`](../trait.Itertools.html#method.map_into) for more information.
+#[must_use = "iterator adaptors are lazy and do nothing unless consumed"]
+pub struct MapInto<I, R> {
+    iter: I,
+    _res: PhantomData<R>,
```

Comment:
Should be `PhantomData<fn() -> R`, we don't own an `R`, we just produce one. Implications for Send/Syncness for example, with this change `MapInto<I, X>` will be `Send` even if `X` is not.